index.js

```diff
@@ -34,39 +35,45 @@ function generateReport(xml) {
   // Remove newlines.
   xml = xml.replace('\n', '');
 
-  report = fromXml.toJson(xml, {object: true, sanitize: false}).resultset;
 
-  // If no problems were found, abort.
-  if (!report.results.result) {
-    return formatReport(report.summary.status, errors, warnings);
-  }
+  parseString(xml, {explicitArray: false}, function(err, data) {
+    if (err) {
+      failedParsing = err;
+    } else {
+      report = data.resultset;
 
-  results = report.results.result instanceof Array
-          ? report.results.result
-          : [report.results.result];
+      if (!report.results.result) {
+        return callback(null, formatReport(report.summary.status, errors, warnings));
+      }
 
-  results.forEach(function(result) {
-    if (ignoreIt(result)) return;
-    if (result.resultType === 'Error') {
-      errors.push({
-        line: result.lineNum,
-        column: result.columnNum,
-        message: getErrorMsg(result.errorMsg),
-        solution: result.repair
-      });
-    }
-    if (result.resultType === 'Potential Problem') {
-      warnings.push({
-        line: result.lineNum,
-        column: result.columnNum,
-        message: getErrorMsg(result.errorMsg),
-        source: result.errorSourceCode
+      results = report.results.result instanceof Array
```

Comment:
What ES environment are we in here. Is [`isArray`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray) available?